txinput.go

```diff
@@ -49,6 +50,41 @@ func NewInputFromBytes(bytes []byte) (*Input, int, error) {
 	}, totalLength, nil
 }
 
+// NewInputFromReader returns a transaction input from the io.Reader provided.
+func NewInputFromReader(r io.Reader) (*Input, error) {
```

Comment:
This also duplicates NewInputFromBytes, if you read r to bytes.Buffer then you can just pass the bytes array to NewInputFromBytes.